### PR TITLE
[CMakeRC] Add new port

### PR DESCRIPTION
**Pull request by Pospelove**
CMakeRC - A Standalone CMake-Based C++ Resource Compiler
https://github.com/vector-of-bool/cmrc

This port contains only one CMake file.

Port is called `cmakerc`, not `cmrc` cause we need `find_package(CMakeRC REQUIRED)`, not `find_package(cmrc REQUIRED)`

Usage

```cmake
find_package(CMakeRC REQUIRED)

# Creates assets library containing all files in `assets` directory
file(GLOB_RECURSE asset_sources assets/*)
cmrc_add_resource_library(assets ${asset_sources})

# Links assets library to our executable
target_link_libraries(main PRIVATE assets)
```

### CODE REVIEW

**Comment by Neumann-A** [comment]
This is not a library why package it with vcpkg?

**Comment by Pospelove** [comment]
> This is not a library why package it with vcpkg?

Actually, it is. The author just embeds C++ code into CMake:
https://github.com/vector-of-bool/cmrc/blob/master/CMakeRC.cmake#L66
https://github.com/vector-of-bool/cmrc/blob/master/CMakeRC.cmake#L442

So in C++, you use it like a normal library.
```c++
#include <cmrc/cmrc.hpp>

int main() {
    // ...
}
```

I'm not sure how to handle such things properly. That's why I created the PR. Looking forward to changes to be requested.

Why I ever created the PR is because I want all dependencies to be managed by vcpkg without copy-pasting anything into sources. In fact, CMakeRC can be treated as a header-only library that is hidden inside a CMake script.

File: ports/cmakerc/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  SHA512 134c358143bd43d38802c5b00b3f5cd6a0e4c7918659dd4cc6726a15048e11376e915889788d0dd6bbfe4c4057e18c9b5306c4e785ad16cb0f4cd1c5c4634f80
+  HEAD_REF master
+)
+
+file(INSTALL "${SOURCE_PATH}/CMakeRC.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME cmakerc-config.cmake)
+
+# Handle copyright
+file(INSTALL "${SOURCE_PATH}/LICENSE.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+
+set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
```
Comment (JackBoosY) [inline ports/cmakerc/portfile.cmake:14]: Please move this line to the top of this file.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by dg0yt** [comment]
> Port is called `cmakerc`, not `cmrc` cause we need `find_package(CMakeRC REQUIRED)`, not `find_package(cmrc REQUIRED)`

FTR the CMake package name doesn't need to match the port name. The config just needs to go a matching directory. However, the project calls itself "CMakeRC". OTOH the C++ functions and includes are in namespace `cmrc`.

**Review by JonLiu1993 — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]
I think it's good to me.

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
There's currently an outstanding issue with the CLA bot; I pushed a merge into this to get the Actions bot to confirm that the version DB edits were OK (they were as far as I could tell).

Because it was entirely green before this (in https://github.com/microsoft/vcpkg/pull/24454/commits/8efad511767216a67d257d35f060fba05867c802 ) I'm going to force merge this once everything else is green.

**Pull request merged**